File: leetcode/python/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = dict()

        for index, num in enumerate(nums):
            # print(index, num)
            if target - num in cache:
                return [cache[target-num], index]
            cache[num] = index

        # print(cache)
        return [-1,-1]

File: leetcode/cpp/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> cache;

        for(int i = 0; i < nums.size(); i++)
        {
            if(cache.contains(target-nums[i]))
            {
                return {cache[target-nums[i]], i};
            }
            cache[nums[i]] = i;
        }
        return {-1,-1};
    }
};